lib/active_merchant/connection.rb

```diff
@@ -82,6 +82,11 @@ def request(method, body, headers = {})
               # very unambiguously does not.
               raise ArgumentError, "DELETE requests do not support a request body" if body
               http.delete(endpoint.request_uri, headers)
+            when :delete_with_body
+              debug body
+              req = Net::HTTP::Delete.new(endpoint.request_uri, headers)
+              req.body = body
+              http.request(req)
```

Comment:
@jasonwebster or @bdewater - we'd appreciate another set of eyes on this bit. This gateway requires a `DELETE` with a body when performing a `refund` or `void`. Any thoughts or concerns here? For some additional history, we almost went with the same approach in https://github.com/activemerchant/active_merchant/pull/2326, but the `delete_with_body` addition to  `lib/active_merchant/connection.rb` didn't land in master.

Reply:
Creating a 'pseudo' HTTP verb here doesn't feel right to me me. It also burdens the caller knowing about ambiguous specs and resulting implementation choices. Is there a strong reason against using these lines to replace the current `:delete` implementation?

Reply:
@bdewater our original implementation had it in the `:delete` but we thought this was safer. I don't think we would be against going back to the original

Reply:
As long as the default behaviour remains the same - not sending a body unless one is specified - I think we should be good. Everybody has programmed around that exception up until now.